src/AutoHashEqualsCached.jl

```diff
@@ -37,25 +54,27 @@ function _show_default_auto_hash_equals_cached(io::IO, @nospecialize(x))
 end
 
 # Find the first struct declaration buried in the Expr.
-get_struct_decl(typ) = nothing
-function get_struct_decl(typ::Expr)
+get_struct_decl(__source__, typ) = nothing
+function get_struct_decl(__source__, typ::Expr)
     if typ.head === :struct
         return typ
     elseif typ.head === :macrocall
-        return get_struct_decl(typ.args[3])
+        return get_struct_decl(__source__, typ.args[3])
     elseif typ.head === :block
         # get the first struct decl in the block
         for x in typ.args
-            if x.head === :macrocall || x.head === :struct || x.head === :block
-                result = get_struct_decl(x)
+            if x isa LineNumberNode
+                __source__ = x
+            elseif x isa Expr && (x.head === :macrocall || x.head === :struct || x.head === :block)
+                result = get_struct_decl(__source__, x)
                 if !isnothing(result)
                     return result
                 end
             end
         end
     end
 
-    error("macro @auto_hash_equals_cached should only be applied to a struct")
+    error("$(__source__.file):$(__source__.line): macro @auto_hash_equals_cached should only be applied to a struct")
```

Comment:
should all of these errors be `ArgumentError` as they are about invalid usage/arguments to the macro? might be a bit more descriptive than an `ExceptionError`

Reply:
Sure.